packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.java

```diff
@@ -216,6 +235,22 @@ public void run() {
     });
   }
 
+  private boolean isDeviceSecure() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      return getKeyguardManager().isDeviceSecure();
+    } else {
+      // NOTE: `KeyguardManager#isKeyguardSecure()` considers SIM locked state,
+      // but it will be ignored on falling-back to device credential on biometric authentication.
+      // That means, setting level to `SECURITY_LEVEL_SECRET` might be misleading for some users.
+      // But there is no equivalent APIs prior to M.
+      // Newer version (>= 1.1.0-alpha01) of `androidx.biometric` library has an introduced
+      // `BiometricManager#canAuthenticate(int)` which will be an alternative of `KeyguardManager#isDeviceSecure()`,
+      // it is not a stable release version yet though.
```

Comment:
Stable version of `androidx.biometric` is out today (version 1.1.0), so it might be a first option.
https://developer.android.com/jetpack/androidx/releases/biometric

It deprecates some methods used in `expo-local-authentication` though. (e.g. [BiometricManager#canAuthenticate()](https://developer.android.com/reference/androidx/biometric/BiometricManager#canAuthenticate()))

I can update and replace some deprecated function usage if that's OK to members of expo :)

Reply:
That would be nice ;) 
What do you think? cc @byCedric

Reply:
@lukmccall @byCedric 
Updated androidx.biometric library at https://github.com/expo/expo/compare/37ad0c987048752ff61bd7df6d86c6001c3d9010...67ba846fddcf2aa475d612196df38ccc0b593776

At first, I thought I could call `BiometricManager#canAuthenticate(int)` with [`BiometricManager.Authenticators#DEVICE_CREDENTIAL`](https://developer.android.com/reference/androidx/biometric/BiometricManager.Authenticators#DEVICE_CREDENTIAL) alone.
But it turned out that `DEVICE_CREDENTIAL` alone was not supported prior to API 30.
So we need to use KeyguardManager#isKeyguardSecure() still.

> Note that not all combinations of authenticator types are supported prior to Android 11 (API 30). Specifically, DEVICE_CREDENTIAL alone is unsupported prior to API 30

https://developer.android.com/reference/androidx/biometric/BiometricManager#canAuthenticate(int)

---

btw I can split PR into API addition and library update, if you want to :)

Reply:
It isn't a big PR so you don't have to split it ;)